File: mobile/lib/features/my_plants/domain/entities/Plant.dart
/// Plant entity representing a microgreen plant
class Plant {
  final int? id;
  final String name;
  final String type; // e.g., "Basil", "Arugula", "Mint", "Cilantro"
  final String? description;
  final DateTime? plantingDate;
  final String growthStage; // e.g., "Seedling", "Growing", "Ready to Harvest", "Harvested"
  final String? healthStatus; // e.g., "Healthy", "Needs Water", "Warning"
  final String? notes;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Plant({
    this.id,
    required this.name,
    required this.type,
    this.description,
    this.plantingDate,
    this.growthStage = 'Seedling',
    this.healthStatus,
    this.notes,
    this.createdAt,
    this.updatedAt,
  });

  /// Factory constructor to convert JSON from backend
  factory Plant.fromJson(Map<String, dynamic> json) {
    return Plant(
      id: json['id'] as int?,
      name: json['name'] as String,
      type: json['type'] as String? ?? json['plantType'] as String? ?? 'Unknown',
      description: json['description'] as String?,
      plantingDate: json['plantingDate'] != null
          ? DateTime.parse(json['plantingDate'] as String)
          : null,
      growthStage: json['growthStage'] as String? ?? 'Seedling',
      healthStatus: json['healthStatus'] as String?,
      notes: json['notes'] as String?,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'] as String)
          : null,
      updatedAt: json['updatedAt'] != null
          ? DateTime.parse(json['updatedAt'] as String)
          : null,
    );
  }

  /// Convert to JSON to send to backend
  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      'name': name,
      'plantType': type, // Backend uses 'plantType' field name
      if (description != null) 'description': description,
      if (plantingDate != null) 'plantingDate': plantingDate!.toIso8601String().split('T')[0], // Send as date only (YYYY-MM-DD)
      'growthStage': growthStage,
      if (healthStatus != null) 'healthStatus': healthStatus,
      if (notes != null) 'notes': notes,
    };
  }

  /// Create a copy with updated fields
  Plant copyWith({
    int? id,
    String? name,
    String? type,
    String? description,
    DateTime? plantingDate,
    String? growthStage,
    String? healthStatus,
    String? notes,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Plant(
      id: id ?? this.id,
      name: name ?? this.name,
      type: type ?? this.type,
      description: description ?? this.description,
      plantingDate: plantingDate ?? this.plantingDate,
      growthStage: growthStage ?? this.growthStage,
      healthStatus: healthStatus ?? this.healthStatus,
      notes: notes ?? this.notes,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  /// Get days since planting
  int? get daysSincePlanting {
    if (plantingDate == null) return null;
    return DateTime.now().difference(plantingDate!).inDays;
  }

  /// Get plant icon based on type
  String get iconEmoji {
    switch (type.toLowerCase()) {
      case 'basil':
        return '🌿';
      case 'arugula':
        return '🥬';
      case 'mint':
        return '🌱';
      case 'cilantro':
        return '🌿';
      case 'lettuce':
        return '🥗';
      case 'spinach':
        return '🥬';
      default:
        return '🌱';
    }
  }
}

File: backend/src/main/java/com/aliparwiz/microgreens/plant/Plant.java
package com.aliparwiz.microgreens.plant;

import com.aliparwiz.microgreens.auth.User;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Plant Entity
 * Represents a microgreen plant tracked by the user
 */
@Entity
@Table(name = "plants")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Plant {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(name = "plant_type", nullable = false)
    private String plantType; // e.g., "Basil", "Arugula", "Mint", "Cilantro"
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(name = "planting_date")
    private LocalDate plantingDate;
    
    @Column(name = "growth_stage")
    private String growthStage = "Seedling"; // "Seedling", "Growing", "Ready to Harvest", "Harvested"
    
    @Column(name = "health_status")
    private String healthStatus; // "Healthy", "Needs Water", "Warning"
    
    @Column(columnDefinition = "TEXT")
    private String notes;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();
    
    // Many plants belong to one user
    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User owner;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
